src/Sentry/GlobalSessionManager.cs

```diff
@@ -234,7 +220,22 @@ private SessionUpdate EndSession(Session session, SessionEndStatus status)
                 return null;
             }
 
-            return EndSession(session, status);
+            return EndSession(session, status, timestamp);
+        }
+
+        public SessionUpdate? ReportError()
+        {
+            if (_currentSession is { } session)
+            {
+                session.ReportError();
+                return session.CreateUpdate(false);
```

Comment:
This could happen after the session is removed from the scope in order to be closed. So technically we could be sending session updates after a closed session.

We're probably safer using a full lock around any of these methods that return `SessionUpdate` to make sure once we close the session, no session updates are created for it.

Reply:
I'm not sure how a lock here would help, because session manager is not responsible for sending session updates, the Hub is.

Reply:
All we can guarantee from here is that the order of calls into this type will return session updates that never result in a session update after a session close. We can do that with locks